internal/controller/deduper/metrics.go

```diff
@@ -0,0 +1,80 @@
+package deduper
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	promNamespace = "buildkite"
+	promSubsystem = "deduper"
+)
+
+// Overridden by New to return len(inFlight).
+var jobsRunningGaugeFunc = func() int { return 0 }
+
+var (
+	_ = promauto.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Subsystem: promSubsystem,
+		Name:      "jobs_running",
+		Help:      "Current number of running jobs according to deduper",
+	}, func() float64 { return float64(jobsRunningGaugeFunc()) })
```

Comment:
With the `_ = ` is this merely a template for adding the rest of the vars below?

Reply:
It's `_` merely because this gauge doesn't need to be referred to within the package later on - `promauto` registers the metric, then the value is obtained during metric scrape through the callback (unlike most of the other metrics, where the value is set or incremented or added or observed on the metric).

Reply:
Excellent, thank you for the explanation!